sql_metadb/derived_tables/feesfines_accounts_actions.sql

```diff
@@ -1,44 +1,96 @@
--- join all transaction data from the feesfines_actions table
--- add patron group information from user_group table
+--Creates a derived table that takes folio_feesfines.accounts as the main table
+--Payments and credits are shown in as a negative balance for accounting purposes
 
 DROP TABLE IF EXISTS feesfines_accounts_actions;
 
 CREATE TABLE feesfines_accounts_actions AS
+
 SELECT
     fa.id AS fine_account_id,
     jsonb_extract_path_text(fa.jsonb, 'amount')::numeric(12,2) AS fine_account_amount,
-    jsonb_extract_path_text(fa.jsonb, 'dateCreated')::timestamptz AS fine_date, 
-    jsonb_extract_path_text(fa.jsonb, 'dateUpdated')::timestamptz AS fine_updated_date,
+    fa.creation_date::timestamptz AS fine_date, 
+    jsonb_extract_path_text(fa.jsonb, 'metadata','updatedDate')::timestamptz AS fine_updated_date,
     jsonb_extract_path_text(fa.jsonb, 'feeFineId')::uuid AS fee_fine_id,
     jsonb_extract_path_text(fa.jsonb, 'ownerId')::uuid AS owner_id,
     jsonb_extract_path_text(fa.jsonb, 'feeFineOwner') AS fee_fine_owner,
     jsonb_extract_path_text(fa.jsonb, 'feeFineType') AS fee_fine_type,
     jsonb_extract_path_text(fa.jsonb, 'materialTypeId')::uuid AS material_type_id,
     jsonb_extract_path_text(fa.jsonb, 'materialType') AS material_type,
-    jsonb_extract_path_text(fa.jsonb, 'payment_status') AS payment_status,
+    jsonb_extract_path_text(fa.jsonb, 'paymentStatus', 'name') AS payment_status,
     jsonb_extract_path_text(fa.jsonb, 'status', 'name') AS fine_status, -- open or closed
     jsonb_extract_path_text(fa.jsonb, 'userId')::uuid AS account_user_id,
     ff.id AS transaction_id,
-    jsonb_extract_path_text(ff.jsonb, 'accountId')::uuid AS account_id,
     jsonb_extract_path_text(ff.jsonb, 'amountAction')::numeric(12,2) AS transaction_amount,
     jsonb_extract_path_text(ff.jsonb, 'balance')::numeric(12,2) AS account_balance,
     jsonb_extract_path_text(ff.jsonb, 'typeAction') AS type_action,
     jsonb_extract_path_text(ff.jsonb, 'dateAction')::timestamptz AS transaction_date,
     jsonb_extract_path_text(ff.jsonb, 'createdAt') AS transaction_location,
     jsonb_extract_path_text(ff.jsonb, 'transactionInformation') AS transaction_information,
-    jsonb_extract_path_text(ff.jsonb, 'source') AS operator_id,
+    jsonb_extract_path_text(ff.jsonb, 'source') AS action_created_by,
     jsonb_extract_path_text(ff.jsonb, 'paymentMethod') AS payment_method,
     uu.id AS user_id,
     uu.patron_group AS user_patron_group_id,
-    ug.id AS patron_group_id,
-    ug.group AS patron_group_name
+    ug.group AS patron_group_name,
+CASE WHEN
+    jsonb_extract_path_text(ff.jsonb, 'typeAction') IN
+    ('Paid partially','Paid fully','Waived partially','Waived fully','Credited partially','Credited fully')
+    THEN jsonb_extract_path_text(ff.jsonb, 'amountAction')::numeric(12,2) * -1
+    ELSE jsonb_extract_path_text(ff.jsonb, 'amountAction')::numeric(12,2)
+    END AS signed_transaction_amount
```

Comment:
Maybe this could be indented so that it will not be at the same indentation level as the `FROM`.

Also there might be some way to avoid the `WHEN`, `THEN`, and `ELSE` all being at the same indentation level.